contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -657,12 +684,29 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
+    Report memory report_ = _decodeReport(report);
```

Comment:
at this point in the code you haven't verified the report, or that the `msg.sender` is pre-verified.

ideally we only pay the gas to decode the report once we have validated the sender as pre-verified OR all of the other aspects of a `transmit` call (including the report itself).

Reply:
Not sure if I'm following, how would we check if the report exists without decoding it?

Reply:
the issue is that the report can still be invalid at this point.

you haven't done `_validateReport` or `msg.sender == s_secondaryProxy`.

i think you will need to move the report validation logic farther up into the function in order to ensure that you are working with a valid report in the first place.

or you could remove the `exist &&` from your `msg.sender == s_secondaryProxy` check, and then call `_decodeReport` inside of that block.

for example:

```solidity
if (msg.sender == s_secondaryProxy) {
  // ensure that the sender is pre-verified so that we can safely decode the report
  Report memory report_ = _decodeReport(report);
  (bool exist, uint32 roundId) = _reportExists(report_);
  ...
} else {
  // leave the signature for `_report` like it was originally
  ...
}```

Reply:
Now I understand, this makes a lot of sense, thanks Kyle!

Modified the code to move all the logic inside the `if` conditional.